rai/node/bootstrap.cpp

```diff
@@ -1547,33 +1551,43 @@ void rai::bulk_pull_server::set_current_end ()
 		}
 		request->end.clear ();
 	}
-	rai::account_info info;
-	auto no_address (connection->node->store.account_get (transaction, request->start, info));
-	if (no_address)
-	{
+
+	if (connection->node->store.block_exists (transaction, request->start)) {
 		if (connection->node->config.logging.bulk_pull_logging ())
 		{
-			BOOST_LOG (connection->node->log) << boost::str (boost::format ("Request for unknown account: %1%") % request->start.to_account ());
+			BOOST_LOG (connection->node->log) << boost::str (boost::format ("Bulk pull request for block hash: %1%") % request->start.to_string ());
 		}
-		current = request->end;
-	}
-	else
-	{
-		if (!request->end.is_zero ())
+
+		current = request->start;
+	} else {
+		rai::account_info info;
+		auto no_address (connection->node->store.account_get (transaction, request->start, info));
+		if (no_address)
 		{
-			auto account (connection->node->ledger.account (transaction, request->end));
-			if (account == request->start)
-			{
-				current = info.head;
-			}
-			else
+			if (connection->node->config.logging.bulk_pull_logging ())
 			{
-				current = request->end;
+				BOOST_LOG (connection->node->log) << boost::str (boost::format ("Request for unknown account: %1%") % request->start.to_account ());
 			}
+			current = request->end;
 		}
 		else
 		{
-			current = info.head;
+			if (!request->end.is_zero ())
+			{
+				auto account (connection->node->ledger.account (transaction, request->end));
+				if (account == request->start)
+				{
+					current = info.head;
+				}
+				else
+				{
+					current = request->end;
+				}
+			}
+			else
+			{
+				current = info.head;
+			}
```

Comment:
Minor, but the last else could be simplified to

```
current = info.head;
if (!request->end.is_zero () 
    && request->start != connection->node->ledger.account (transaction, request->end))
{
	current = request->end;
}
```

Reply:
or even
```cpp
if (!request->end.is_zero () 
    && request->start != connection->node->ledger.account (transaction, request->end))
{
	current = request->end;
}
else
{
	current = info.head;
}
```